app/routes/_auth+/onboarding.tsx

```diff
@@ -160,27 +160,25 @@ export default function OnboardingPage() {
 					<Field
 						labelProps={{ htmlFor: fields.username.id, children: 'Username' }}
 						inputProps={{
-							...fields.username,
+							...conform.input(fields.username),
 							autoComplete: 'username',
-							autoFocus: true,
+							autoFocus: typeof actionData === 'undefined' || typeof fields.username.initialError !== 'undefined',
```

Comment:
The conform helper will set the `autoFocus` attribute for us if there is any error in the previous submission. This overrides it so that we can have the field autofocus by default, but not if there is a previous submission with no error on the username field.